benches/arrays/sort.ncl

```diff
@@ -1,5 +1,27 @@
-let nums = [3288, 2438, 1462, 437, 17, 1929, 2595, 2702, 831, 3685, 1906, 420, 2646, 1906, 2539, 184, 299, 3455, 2320, 1471, 1812, 3837, 1000, 2097, 945, 1973, 1090, 2285, 2272, 1045, 848, 3194, 2113, 537, 1953, 561, 2082, 2416, 3182, 3660, 3379, 1458, 39, 2720, 2469, 811, 837, 548, 3021, 2961, 2860, 2092, 942, 2511, 1769, 168, 130, 2755, 2552, 1057, 2619, 1848, 2373, 2012, 2339, 2521, 3593, 2784, 3094, 2260, 1994, 690, 828, 1298, 3434, 2929, 2188, 1285, 3783, 3635, 971, 3334, 87, 1578, 1363, 533, 3634, 1874, 830, 3976, 1140, 789, 1483, 1966, 1626, 3552, 2968, 3301, 749, 2576] in
-let ascending = fun x y => if x < y then `Less else if (x == y) then `Equal else `Greater in
-{
-    run = fun n => array.sort ascending nums
-}
+let mgc 
+  | doc "A multiplicative congruential pseudorandom number generator (MCG)."
+  = fun params x => (params.a * x + params.c) % params.m in
+
+# NOTE: These parameters are from "Table 7: Good multipliers for MCGs with m = 2^64"
+# of the 2021 paper "Computationally Easy, Spectrally Good Multipliers for
+# Congruential Pseudorandom Number Generators", by Guy Steele and Sebastiano Vigna.
+# But, they cannot be used because they require bit-shifting to select the 32 MSBs.
+# let params = { m = num.pow 2 32, a = 2480367069 } in
+
+# These parameters are from the "Numerical Recipes" book.
+let params = { m = num.pow 2 32, a = 1664525, c = 1013904223 } in
+
+let rec array_random = fun init n =>
+  if n == 0 then [init]
+  else 
+    let next = mgc params init in
+    [next] @ array_random next (n - 1)
+in
+
+let ascending = fun x y => 
+  if x < y then `Lesser 
+  else if x == y then `Equal 
+  else `Greater 
+in
+
+{ run = fun n => array.sort ascending (array_random 1337 n) }
```

Comment:
The code is rather simple, a good surprise. But the issue is now that we don't benchmark sorting, we benchmark random array generation + sorting. One possibility is to do that on the rust side, and then pass it to the `run` function. Or make a custom function to run this benchmark which first evaluate `array_random 1337 n`, before starting the measurement, and then pass it to run?

But now that I write it, the latter seems to imply the same pain of writing a custom benchmark function, and is just more complicated. Maybe the simplest is to generate the array on the Rust side and pass it to the run function instead of `n`.

Reply:
That's fair. Maybe I could also split the number generation stuff into its own benchmark?

Reply:
Yes, since the code is written, let's add it

Reply:
Should be all good now. 

Though I used mostly the same code on the Rust side to avoid adding `rand` to the dependencies. We probably already have it indirectly but this is such a minor requirement.